onmt/transforms/fuzzymatching.py

```diff
@@ -0,0 +1,149 @@
+from onmt.utils.logging import logger
+from onmt.transforms import register_transform
+from .transform import Transform
+
+from rapidfuzz import fuzz, process
+import numpy as np
+import time
+
+_FUZZY_TOKEN = '｟fuzzy｠'
+
+
+class FuzzyMatcher(object):
+    """Class for creating and setting up fuzzy matchers."""
+
+    def __init__(self, tm_path, corpus_ratio, threshold=70, tm_delimiter='\t'):
+        self.threshold = threshold
+        self.corpus_ratio = corpus_ratio
+        self.tm_delimiter = tm_delimiter
+        self.internal_tm = self._create_tm(tm_path)
+
+    def _create_tm(self, tm_path):
+        src_segments, tgt_segments = list(), list()
+        with open(tm_path, mode='r', encoding='utf-8') as file:
+            pairs = file.readlines()
+            for pair in pairs:
+                source, target = map(str, pair.split(self.tm_delimiter))
+
+                # Filter out very short or very long sentences
+                # from the TM for better performance
+                if len(source) < 4 or len(source) > 70:
+                    continue
+                src_segments.append(source.strip())
+                tgt_segments.append(target.strip())
+        logger.info(f'Translation Memory size for fuzzymatching transform: '
+                    f'{len(src_segments)}')
+        return [src_segments, tgt_segments]
+
+    def _get_batch_matches(self, batch):
+        logger.info(f'Starting fuzzy matching on {len(batch)} examples')
+        fuzzy_count = 0
+        start = time.time()
+        augmented = list()
+
+        # We split the `batch` and perform fuzzy matching
+        # in smaller batches in order to reduce memory usage.
+        # Perfomance is not affected.
+        portion = 25
```

Comment:
This may depend on the available RAM and the `bucket` options in the config, but I couldn't think of a formula to make it safe and generic so I made only safe since matching speed does not seem to be affected. With the default bucket size (~250k), 64 GB RAM, "mini_batches" of 10k, and a TM with ~200k pairs, it seems to keep RAM stressed but under control.
If you have an ideas, please suggest.

Reply:
Why not something more explicit like setting some `fuzzy_match_chunk_size`, with a default of 10k (which seems to be an okay value from what you describe)?
(Might be a bit overkill though, we can probably keep the set portion at first.)

Reply:
I made this part a bit more explicit, but I also think it's too much to give an option for that.